spec/factories/matches.rb

```diff
@@ -22,10 +22,15 @@
     association :team_away, factory: :team
     association :team_home, factory: :team
     association :event, factory: :event
-    score_home { rand(10..20) }
-    score_away { rand(1..9) }
     gameday 0
     points_away 3
     points_home 1
+
+    factory :match_with_results do
```

Comment:
Have a look at rspec traits

Reply:
I think this is not doable because the callback `after(:create)` is used